### ISSUE TITLE
no-extra-parens false positive with TypeScript `as`

### ISSUE THREAD

**Issue by bcuccioli** [eslint/eslint#13604]
**Tell us about your environment**

* Node version: v13.14.0
* npm version: v6.14.5
* Local ESLint version: v7.2.0 (Currently used)

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const x: any[] = [];
(x as string[]).push('hi');
```

Lint command:

```bash
$ ./node_modules/.bin/eslint src/test.ts
```

**What did you expect to happen?**

No lint warnings

**What actually happened? Please include the actual, raw output from ESLint.**

```
3:1  error  Unnecessary parentheses around expression  no-extra-parens
```

**Are you willing to submit a pull request to fix this bug?**

No
